### DISCUSSION TITLE
Adhere to excludedSchemas List to Limit Functions Returned

### DISCUSSION THREAD

**Discussion by w3b6x9** [supabase/supabase#13773] [Feature Requests]
# Feature request

## Is your feature request related to a problem? Please describe.

Studio Functions returns a list of all database functions, even the ones belonging to schemas that are excluded.

## Describe the solution you'd like

Pass a list of excluded schemas when fetching a list of database functions.

## Additional context

Related: https://github.com/supabase/postgres-meta/issues/178

**Comment by joshenlim** [comment]
some contextual info - we do have `excludedSchemas` in studio, but this is purely client side

what happens is the studio fetches _all_ of the data, then filters out based on `excludedSchemas` - as mentioned in the related issue, probably need this on the BE side instead

**Comment by w3b6x9** [comment]
@joshenlim yep, will need this in pg-meta but then we'll need to update the pg-meta functions api in studio to pass in `excludedSchemas`.

**Comment by sweatybridge** [comment]
fyi, the pg-meta ticket is unblocked now. https://github.com/supabase/postgres-meta/pull/375

**Comment by MildTomato** [comment]
Can we get more context here @w3b6x9 ?

What do you actually want? To see all the functions in Studio UI ?

**Reply by w3b6x9** [reply]
I think this is outdated now and doesn't apply anymore.
